fedimint-core/src/db/mod.rs

```diff
@@ -1930,11 +1930,26 @@ pub async fn apply_migrations(
     target_db_version: DatabaseVersion,
     migrations: BTreeMap<DatabaseVersion, CoreMigrationFn>,
     module_instance_id: Option<ModuleInstanceId>,
+    // When used in client side context, we can/should ignore keys that external app
+    // is allowed to use, and but since this function is shared, we make it optional argument
+    external_prefixes_above: Option<u8>,
 ) -> Result<(), anyhow::Error> {
     // Newly created databases will not have any data since they have just been
     // instantiated.
     let mut dbtx = db.begin_transaction_nc().await;
-    let is_new_db = dbtx.raw_find_by_prefix(&[]).await?.next().await.is_none();
+    let is_new_db = dbtx
+        .raw_find_by_prefix(&[])
+        .await?
+        .filter(|(key, _v)| {
+            std::future::ready(
+                external_prefixes_above.map_or(true, |external_prefixes_above| {
```

Comment:
nit: move this to if let outside the filter, we don't have to pay the checking cost on server side

i would be really suprised if compiler is able optimize through streams wrapper

Reply:
True, but comparing to the IO this really doesn't matter. My guestimate is orders of magnitude (1ns vs 1us) differences. And this code runs ones on start anyway.